lib/apollo-client.js

```diff
@@ -6,22 +6,50 @@ import { setContext } from '@apollo/client/link/context';
 import { onError } from '@apollo/client/link/error';
 
 import { getFromLocalStorage, LOCAL_STORAGE_KEYS } from './local-storage';
-import { getGraphqlUrl } from './utils';
 
-let apolloClient = null;
+let apolloClient, customAgent;
 
-const graphqlFetch = async (url, options = {}) => {
-  // Add headers to help the API identify origin of requests
-  if (!process.browser) {
-    options.headers = options.headers || {};
-    options.headers['oc-env'] = process.env.OC_ENV;
-    options.headers['oc-secret'] = process.env.OC_SECRET;
-    options.headers['oc-application'] = process.env.OC_APPLICATION;
-    options.headers['user-agent'] = 'opencollective-frontend/1.0 node-fetch/1.0';
+const getBaseApiUrl = () => {
+  if (process.browser) {
+    return '/api';
   }
 
+  return process.env.INTERNAL_API_URL || process.env.API_URL || 'https://api.opencollective.com';
+};
+
+/**
+ * Returns the GraphQL api url for the appropriate api version and environment.
+ * @param {string} version - api version. Defaults to v1.
+ * @returns {string} GraphQL api url.
+ */
+const getGraphqlUrl = apiVersion => {
+  const apiKey = !process.browser ? process.env.API_KEY : null;
+  return `${getBaseApiUrl()}/graphql${apiVersion ? `/${apiVersion}` : ''}${apiKey ? `?api_key=${apiKey}` : ''}`;
+};
+
+const getCustomAgent = () => {
+  if (!customAgent) {
+    const http = require('http');
+    const https = require('https');
+    const httpAgent = new http.Agent({ keepAlive: true });
+    const httpsAgent = new https.Agent({ keepAlive: true });
```

Comment:
Would be nice to have an expiration on the keepAlive (2min? 5min?) to handle certificate rotation. Unless it's done automatically?